src/coreclr/src/vm/proftoeeinterfaceimpl.cpp

```diff
@@ -8827,15 +8827,15 @@ HRESULT ProfToEEInterfaceImpl::GetReJITIDs(
         // The rejit tables use a lock
         CAN_TAKE_LOCK;
 
-
         PRECONDITION(CheckPointer(pcReJitIds, NULL_OK));
         PRECONDITION(CheckPointer(reJitIds, NULL_OK));
+        PRECONDITION((cReJitIds == 0) == (reJitIds == NULL));
 
     }
     CONTRACTL_END;
 
     PROFILER_TO_CLR_ENTRYPOINT_SYNC_EX(
-        kP2EEAllowableAfterAttach,
+        kP2EEAllowableAfterAttach | kP2EETriggers,
```

Comment:
I think this change is for the best, but its possible that it will generate an error in someone's profiler if they were previously calling the API from a no-trigger callback. I don't recall if we already have some good documentation around which API calls are legal from which callbacks? I was going to say we should raise the issue on the profiler announcement page, but without the rules documented it might raise a bunch of questions we weren't prepared to answer yet.

Reply:
I haven't seen the rule documented anywhere, even on David Broman's blog